lib/puppet/provider/base_dsc/powershell.rb

```diff
@@ -67,8 +71,67 @@ def destroy
     output = powershell(powershell_args, script_content)
     Puppet.debug "Destroy Dsc Resource returned: #{output}"
     data = JSON.parse(output)
+
     fail(data['errormessage']) if !data['errormessage'].empty?
-    true
+
+    notify_status if data['rebootrequired'] == true
+
+    data
+  end
+
+  def notify_status
+    Puppet.info "A reboot is required to progress further. Notifying Puppet."
+
+    reboot_resource = resource.catalog.resource(:reboot, 'dsc_reboot')
+    unless reboot_resource
+      Puppet.warning "No reboot resource found in the graph that has 'dsc_reboot' as it's name. Cannot signal reboot to Puppet."
+      return
+    end
+
+    edge = Puppet::Relationship.new(resource, reboot_resource)
+
+    # if an edge already exists from Reboot[dsc_reboot] to this resource, we have problems
+    if resource.catalog.relationship_graph.edge?(edge.target, edge.source)
+      # TODO: should this be a hard fail? probably?
+      Puppet.warning "Reboot resource 'dsc_reboot' is already connected to #{resource}"
+      return
+    end
+
+    # prevent addition of the same edge in the graph if a manual one exists
+    if !resource.catalog.relationship_graph.edge?(edge.source, edge.target)
+      edge.callback = :refresh
+      edge.event = :ALL_EVENTS
+      resource.catalog.relationship_graph.add_edge(edge)
```

Comment:
At this point we receive a message stating that our dsc_reboot resource has a negative number of dependencies. This comes from puppet relationshipgraph on line [73](https://github.com/puppetlabs/puppet/blob/master/lib/puppet/graph/relationship_graph.rb#L73). I don't understand why this method does this, as `> 0' misses 0 as a non negative number.